superset-frontend/src/components/ListView/utils.ts

```diff
@@ -63,11 +68,11 @@ function updateInList(list: any[], index: number, update: any): any[] {
   ];
 }
 
-function mergeCreateFilterValues(list: Filter[], updateList: FilterValue[]) {
+function mergeCreateFilterValues(list: Filter[], updateObj: any) {
   return list.map(({ id, operator }) => {
-    const update = updateList.find(obj => obj.id === id);
+    const update = updateObj[id] || [];
 
-    return { id, operator, value: update?.value };
+    return { id, operator, value: update[0] };
```

Comment:
so it looks like we're only using the value portion of the query filter state. We could take it further and only serialize the value and not the operator